docs/source/usersguide/input.rst

```diff
@@ -558,19 +559,127 @@ The following code snippet illustrates the instantiation of the ``CPUSolver`` fo
 Fixed Source Calculations
 -------------------------
 
-aljlkjadf
+It is also possible to add a fixed source to any region in OpenMOC. During computation of the total source, a fixed source is added together with the calculated scattering source and fission source for each flat source region. By default, the fixed source is set to zero everywhere. The ``setFixedSourceByFSR(...)`` routine allows the user to set the fixed source in a given flat source region using its unique ID. For most calcuations, setting the fixed source for every flat source region of interest individually can be cumbersome. In addition, this would require retreiving the unique ID for every flat source region in which the user desires to set the fixed source. Therefore, the ``setFixedSourceByCell(...)`` routine allows the user to set the fixed source for every flat source region within a cell to a common value. An example is given below for setting the fixed source of a ``Cell`` with a point source to unity in the first energy group.
+
+.. code-block:: python
+
+  # Set the fixed source inside the source cell in group 1 to unity
+  solver.setFixedSourceByCell(source_cell, 1, 1.0)
+
+
+The equivalent code for setting the point source of all flat source regions within the source cell using ``setFixedSourceByFSR(...)`` is given below.
+
+.. code-block:: python
+
+  # Get the unique ID of the cell containing the point source
+  point_source_cell_id = source_cell.getId()
+
+  # Loop over all FSRs and test if the FSR is within the point source cell
+  for fsr_id in xrange(solver.getGeometry().getNumFSRs()):
+    cell = solver.getGeometry().findCellContainingFSR(fsr_id)
+
+    # If ithe FSR is within the point source cell, set the fixed source
+    if cell.getId() == point_source_cell_id:
+      solver.setFixedSourceByFSR(fsr_id, 1, 1.0)
+
+In this case, it is far simpler to set the fixed source by ``Cell``. However, there may be cases where the user may wish to set the fixed source within a ``Cell`` to different values. For instance, if the user wishes to model a continuously varying fixed source and there are multiple flat source regions within some ``Cell``, then for each flat source region within the cell the fixed source would need to be set individually.
+
+TODO: make fixed source problem
+
+So far only the ``computeEigenvalue(...)`` routine has been introduced for solving neutron transport problems. The OpenMOC ``Solver`` has other solution options in addition to the eigenvalue solver which can be very useful for fixed source calculations. Specifically, the ``computeFlux(...)`` and ``computeSource(...)`` routines solve neutron transport over the problem without computing an eigenvalue.
```

Comment:
Should the first sentence be removed? It sounds you are saying there is only a fixed source solver. Then you say OpenMOC has other solvers.